### ISSUE TITLE
"pkg.installed: -pkgs" may takes a very long time to run "repoquery --all"

### ISSUE THREAD

**Issue by wangyang616** [saltstack/salt#28996]
When I use "pkg.installed" with "pkgs" to install multiple packages, salt run 

`repoquery --plugins --queryformat '%{NAME}_|-%{VERSION}_|-%{RELEASE}_|-%{ARCH}_|-%{REPOID}'   --all --quiet --whatprovides`

But it takes a very long time with large repos cause it fetched all rpms.

Example:

```
# time repoquery --plugins --queryformat '%{NAME}_|-%{VERSION}_|-%{RELEASE}_|-%{ARCH}_|-%{REPOID}'   --all --whatprovides
...
real    0m28.860s
user    0m27.894s
sys     0m0.698s
```

Is it possible to check only specific pkgs?
like: `repoquery --plugins --queryformat '%{NAME}_|-%{VERSION}_|-%{RELEASE}_|-%{ARCH}_|-%{REPOID}' --whatprovides nginx httpd ...`

**Comment by RuriRyan** [comment]
This should have been fixed by #28526.

**Comment by jfindlay** [comment]
@wangyang616, thanks for the report.  As @RuriRyan indicates, this is a duplicate of #28526.  The fix will be included in 2015.8.2
